src/main/java/org/killbill/billing/plugin/adyen/client/payment/builder/checkout/CheckoutPaymentsBuilder.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2014-2018 Groupon, Inc
+ * Copyright 2014-2018 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.plugin.adyen.client.payment.builder.checkout;
+
+import com.adyen.model.checkout.DefaultPaymentMethodDetails;
+import com.adyen.model.checkout.LineItem;
+import com.adyen.model.checkout.PaymentMethodDetails;
+import com.adyen.model.checkout.PaymentsRequest;
+import com.google.common.base.Charsets;
+import com.adyen.model.*;
+import java.util.Base64;
+
+import org.killbill.billing.plugin.adyen.api.mapping.klarna.PropertyMapper;
+import org.killbill.billing.plugin.adyen.client.model.PaymentData;
+import org.killbill.billing.plugin.adyen.client.model.UserData;
+import org.killbill.billing.plugin.adyen.client.model.paymentinfo.KlarnaPaymentInfo;
+import org.killbill.billing.plugin.adyen.client.payment.builder.RequestBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CheckoutPaymentsBuilder extends RequestBuilder<PaymentsRequest> {
+    private final String merchantAccount;
+    private final PaymentData paymentData;
+    private final UserData userData;
+
+    private static final Logger logger = LoggerFactory.getLogger(CheckoutPaymentsBuilder.class);
+    public final static String OPEN_INVOICE_MERCHANT_DATA = "openinvoicedata.merchantData";
+
+    public CheckoutPaymentsBuilder(final String merchantAccount,
+                                   final PaymentData paymentData,
+                                   final UserData userData) {
+        super(new PaymentsRequest());
+        this.merchantAccount = merchantAccount;
+        this.paymentData = paymentData;
+        this.userData = userData;
+    }
+
+    @Override
+    public PaymentsRequest build() {
+        KlarnaPaymentInfo paymentInfo = null;
+        if (paymentData.getPaymentInfo() instanceof KlarnaPaymentInfo) {
+            paymentInfo = (KlarnaPaymentInfo) paymentData.getPaymentInfo();
+        } else {
+            logger.error("Invalid paymentInfo object, expected KlarnaPaymentInfo");
+            return request;
+        }
+
+        if(merchantAccount != null) {
+            request.setMerchantAccount(merchantAccount);
+        } else {
+            logger.error("merchantAccount is null, can not create payment request");
+            return request;
```

Comment:
We are not raising any exception. This error log is deceptive.
What should be the behaviour ?